components/camel-openapi-java/src/main/java/org/apache/camel/openapi/RestOpenApiReader.java

```diff
@@ -358,36 +358,40 @@ private void parseOas30(OpenAPI openApi, RestDefinition rest, String[] pathAsTag
             for (RestSecurityDefinition def : sd.getSecurityDefinitions()) {
                 if (def instanceof BasicAuthDefinition) {
                     SecurityScheme auth = new SecurityScheme().type(SecurityScheme.Type.HTTP)
-                            .scheme("basic").description(def.getDescription());
-                    openApi.getComponents().addSecuritySchemes(def.getKey(), auth);
+                            .scheme("basic").description(CamelContextHelper.parseText(camelContext, def.getDescription()));
+                    openApi.getComponents().addSecuritySchemes(CamelContextHelper.parseText(camelContext, def.getKey()), auth);
                 } else if (def instanceof BearerTokenDefinition) {
                     SecurityScheme auth = new SecurityScheme().type(SecurityScheme.Type.HTTP)
-                            .scheme("bearer").description(def.getDescription())
-                            .bearerFormat(((BearerTokenDefinition) def).getFormat());
-                    openApi.getComponents().addSecuritySchemes(def.getKey(), auth);
+                            .scheme("bearer").description(CamelContextHelper.parseText(camelContext, def.getDescription()))
+                            .bearerFormat(
+                                    (CamelContextHelper.parseText(camelContext, ((BearerTokenDefinition) def).getFormat())));
+                    openApi.getComponents().addSecuritySchemes(CamelContextHelper.parseText(camelContext, def.getKey()), auth);
                 } else if (def instanceof ApiKeyDefinition) {
                     ApiKeyDefinition rs = (ApiKeyDefinition) def;
                     SecurityScheme auth = new SecurityScheme().type(SecurityScheme.Type.APIKEY)
-                            .name(rs.getName()).description(def.getDescription());
-
-                    if (rs.getInHeader() != null && Boolean.parseBoolean(rs.getInHeader())) {
+                            .name(CamelContextHelper.parseText(camelContext, rs.getName()))
+                            .description(CamelContextHelper.parseText(camelContext, def.getDescription()));
+                    String inHeader = CamelContextHelper.parseText(camelContext, rs.getInHeader());
```

Comment:
You can use parseBoolean instead of parseText here

Reply:
Thanks for the review, I have updated the PR.